presto-hive/src/main/java/io/prestosql/plugin/hive/HiveWriterFactory.java

```diff
@@ -549,6 +549,37 @@ else if (insertExistingPartitionsBehavior == InsertExistingPartitionsBehavior.ER
                 hiveWriterStats);
     }
 
+    private String computeFileName(OptionalInt bucketNumber)
+    {
+        if (isCreateTransactionalTable) {
+            // Hive is very opinionated on naming pattern for "original" files of transactional tables.
+            // As currently CTAS for transactional tables in Presto creates non-transactional ("original") files.
+            // we use special file naming mode in this case.
+            //
+            // For bucketed tables we drop query id from file names and just leave <bucketId>_0
+            // For non bucketed tables we use 000000_<uuid_as_number>
+            if (bucketNumber.isPresent()) {
+                return computeBucketedFileName(Optional.empty(), bucketNumber.getAsInt());
+            }
+            else {
+                String paddedBucket = Strings.padStart("0", BUCKET_NUMBER_PADDING, '0');
+                UUID uuid = randomUUID();
+                return format("0%s_%s%s",
+                        paddedBucket,
+                        UnsignedLong.fromLongBits(uuid.getLeastSignificantBits()).toString(),
+                        UnsignedLong.fromLongBits(uuid.getMostSignificantBits()).toString());
+            }
+        }
+        else {
```

Comment:
https://github.com/prestosql/presto/pull/4495

Reply:
Changed, though it kinda looks worse with long code blocks imo.